### ISSUE TITLE
Some arcs randomly printed as full circles.

### ISSUE THREAD

**Issue by Snargaloo** [FormerLurker/ArcWelderPlugin#111]
I printed a model welded with an older version of ArcWelder in PETG for tolerance testing.  When I tried printing this model again in carbon filled PA using a devel branch of ArcWelder, some of the arcs sporadically printed as full circles.  Here is a picture of the PETG part (Orange) next to the PA-CF part:

![IMG_3133](https://user-images.githubusercontent.com/35717808/103419918-ee70d580-4b49-11eb-913a-ac7673647e77.jpg)

Here are some close up pictures of the issue:

![IMG_3127](https://user-images.githubusercontent.com/35717808/103419966-2546eb80-4b4a-11eb-8fa8-c63de2ebcc12.jpg)
![IMG_3128](https://user-images.githubusercontent.com/35717808/103419989-33950780-4b4a-11eb-9c15-014bca0e3695.jpg)
![IMG_3129](https://user-images.githubusercontent.com/35717808/103420003-3db70600-4b4a-11eb-9221-59d7f94155aa.jpg)
![IMG_3130](https://user-images.githubusercontent.com/35717808/103420013-46a7d780-4b4a-11eb-9f92-ed5c84478217.jpg)
![IMG_3131](https://user-images.githubusercontent.com/35717808/103420028-4f98a900-4b4a-11eb-98d6-6d84aa450372.jpg)
![IMG_3132](https://user-images.githubusercontent.com/35717808/103420042-59baa780-4b4a-11eb-9001-df92ec5ef231.jpg)

I am using version Arc Welder v1.0.1rc1.dev2+u.2e83905.  I sliced these models with Cura.  I build Cura from source, and this model was sliced from a build from 4.8 Origin/Master commit 9efe5dd5e29f36732148731c389c8bb280ab2402.

This is the original model gcode:
[HEAD_FRAME_eSun_ePA-CF_FR77651-.zip](https://github.com/FormerLurker/ArcWelderPlugin/files/5757918/HEAD_FRAME_eSun_ePA-CF_FR77651-.zip)

This is the ArcWelded gcode:
[HEAD_FRAME_eSun_ePA-CF_FR77651-.aw.zip](https://github.com/FormerLurker/ArcWelderPlugin/files/5757923/HEAD_FRAME_eSun_ePA-CF_FR77651-.aw.zip)

This is the ArcWelded gcode for the previous ArcWelder version (PETG):
[CE3_HEAD_FRAME.aw.zip](https://github.com/FormerLurker/ArcWelderPlugin/files/5757924/CE3_HEAD_FRAME.aw.zip)

Since this model was printed solid, I decided to trim the extra arc bits and use it.  Please let me know if I can provide any other information.

**Comment by FormerLurker** [comment]
Hi there!  I think this is a duplicate issue that has been solved in the [latest devel rc](https://github.com/FormerLurker/ArcWelderPlugin/releases/tag/1.0.1rc1.dev2).  See the release notes for details, and install with this link (there was a compiler issue with the previous version):  

```https://github.com/FormerLurker/ArcWelderPlugin/archive/2dedd1b9dfa5da4e75e968107473a8113369e709.zip```

I ran your gcode through the new version, and those bad arcs were eliminated.  If you want, send me your processed gcode once you install the new version and I'll run it through my analyzer to verify it's gone for you too.

TLDR:  The path tolerance percent was not being converted from a percent (say 5%) to a decimal (0.05).  This caused issues obviously, and the bug was likely introduced during a refactor.  However, not only did I fix that problem, but I added an algorithm that can detect these issues no matter what the path tolerance percent is (I used 10000% for testing).  It does slow the arc generation down somewhat, but it is the only foolproof way I've come up with for verifying the arcs without setting some kind of arbitrary limits (like the path tolerance percent).

Let me know what you think!

**Comment by Snargaloo** [comment]
I have now installed Arc Welder v1.0.1rc1.dev2+u.a0f41e1.  I have not modified any other software versions and attach the original sliced gcode:
[HEAD_FRAME_eSun_ePA-CF_FR77651-(1).zip](https://github.com/FormerLurker/ArcWelderPlugin/files/5759193/HEAD_FRAME_eSun_ePA-CF_FR77651-.1.zip)

and the welded gcode:
[HEAD_FRAME_eSun_ePA-CF_FR77651-.aw(1).zip](https://github.com/FormerLurker/ArcWelderPlugin/files/5759195/HEAD_FRAME_eSun_ePA-CF_FR77651-.aw.1.zip)

Please process these through your analyzer.  I am about to start some prints with the new version and will report back.  Thank you for the swift response.

**Comment by FormerLurker** [comment]
It looks good now!  Here is what the welded code looks like in Simplify 3D:
![image](https://user-images.githubusercontent.com/33363128/103443366-40862980-4c24-11eb-8305-29454d327d14.png)

You can also install the *PrettyGCode* OctoPrint plugin to view welded files now, as long as you have the latest update.  Here is what it looks like in there:

![image](https://user-images.githubusercontent.com/33363128/103443399-935fe100-4c24-11eb-99c4-35ca9c2155fc.png)

I assisted updating that plugin to render arcs, and it uses a variant of Marlin's arc interpolation algorithm, so it should be quite accurate.

Thanks, and let me know if the print works out!

**Comment by Snargaloo** [comment]
Is the version of Pretty GCode that you mention on the regular versioning branch?  Or is it a development version?

**Comment by Snargaloo** [comment]
I just checked with the version of Pretty GCode that I have installed.  I loaded the welded gcode file that has the full circles and it renders the file just as it printed with full circles!  Thank you for the heads up!

**Comment by FormerLurker** [comment]
@Snargaloo, have you tried the new arcwelder that I linked to (installed in plugin manager->get more->from url... with this link ```https://github.com/FormerLurker/ArcWelderPlugin/archive/2dedd1b9dfa5da4e75e968107473a8113369e709.zip```)?

**Comment by Snargaloo** [comment]
Everything seems to be working well again, thank you!

**Comment by FormerLurker** [comment]
woot!  Thanks for following back up, and enjoy Arc Welder!
